### PR TITLE
RDM-12549: Migrate internal search APIs to Access Profiles

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-12549

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java [hunk 1]
```diff
@@ -93,7 +92,12 @@ private boolean filterDistinctFieldsByRole(final HashSet<String> addedFields, fi
         if (addedFields.contains(id)) {
             return false;
         } else {
-            if (StringUtils.isEmpty(resultField.getRole()) || userRepository.anyRoleEqualsTo(resultField.getRole())) {
+            Set<AccessProfile> accessProfiles =  caseDataAccessControl
+                .generateAccessProfilesByCaseTypeId(resultField.getCaseTypeId());
+            Set<String> accessProfileNames = AccessControlService.extractAccessProfileNames(accessProfiles);
+
+            if (StringUtils.isEmpty(resultField.getRole())
```
[the remark this answers is not in the document]
Comment (danlysiak) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java:99]: As well as the method in this class where these changes are being made - `filterDistinctFieldsByAccessProfile()`

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/SearchQueryOperation.java [hunk 2]
```diff
@@ -127,7 +128,16 @@ private boolean hasSortField(SearchResultField searchResultField) {
...
+        Set<AccessProfile> accessProfiles =  caseDataAccessControl
+            .generateAccessProfilesByCaseTypeId(resultField.getCaseTypeId());
+        Set<String> accessProfileNames = AccessControlService.extractAccessProfileNames(accessProfiles);
+
+        return StringUtils.isEmpty(resultField.getRole())
+            || anyRoleEqualsTo(accessProfileNames, resultField.getRole());
+    }
+
+    private boolean anyRoleEqualsTo(Set<String> accessProfiles, String accessProfile) {
+        return accessProfiles.contains(accessProfile);
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/SearchQueryOperation.java:140]: Just the same comments as from `MergeDataToSearchResultOperation`, and same for next class..

**Review by danlysiak — COMMENTED** [review]
Looks good, just a couple of minors - assume this is backward compatible?

**Review by rafalkalita — APPROVED** [review]
LGTM

**Review by danlysiak — APPROVED** [review]

**Pull request merged**